tableauserverclient/server/query.py

```diff
@@ -31,12 +35,16 @@ def __getitem__(self, k):
         page = self.page_number
         size = self.page_size
 
+        # Create a range object for quick checking if k is in the cached result.
         page_range = range((page - 1) * size, page * size)
 
         if isinstance(k, slice):
```

Comment:
is slice ever set here?

Reply:
`k` can either be an int or a `slice` object.

It could be a slice if it was called in the following manner:

```python
server.workbooks.all()[10:40:3]
```

Reply:
In that example, if the client code calls

```python
server.workbooks.all()[10:40:3]
```

Then `__getitem__` receives `k=slice(start=10, stop=40, step=3)`. Any of the typical slicing syntax will work. For `server.workbooks.all()[:50]` then `k=slice(start=None, stop=50, step=None)`, then the `None`'s get replaced with sensible defaults so that it acts the way that slicing would on a list.

If start is `None`, it becomes `0`
If stop is `None`, it becomes the total number available
If step is `None`, it becomes 1